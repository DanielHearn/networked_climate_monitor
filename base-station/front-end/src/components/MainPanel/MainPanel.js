// Displays a main interface panel
export default {
  name: 'main-panel',
  props: {}
}
