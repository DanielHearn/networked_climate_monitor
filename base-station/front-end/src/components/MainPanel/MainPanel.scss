@import "./../../scss/variables.scss";

.main-panel {
  display: flex;
  flex-direction: column;
  flex: 1 auto;
  width: 100%;
  height: 100%;
  box-sizing: border-box;
  z-index: 10;
  background: $grey-light;
  .main-panel-header {
    padding: 0.5em;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    .heading {
      margin: 0 0.5em;
    }
  }
  .main-panel-content {
    padding: 1em;
  }
}