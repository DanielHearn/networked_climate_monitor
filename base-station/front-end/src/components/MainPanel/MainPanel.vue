<template>
  <div class="main-panel">
    <div class="main-panel-header">
      <slot name="header"></slot>
    </div>
    <div class="main-panel-content">
      <slot name="content"></slot>
    </div>
  </div>
</template>

<script src="./MainPanel.js"></script>

<style lang="scss" src="./MainPanel.scss"></style>
