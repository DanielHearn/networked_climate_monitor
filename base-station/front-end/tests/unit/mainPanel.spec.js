import { mount } from '@vue/test-utils'
import MainPanel from './../../src/components/MainPanel/MainPanel.vue'

describe('MainPanel.vue', () => {
  let wrapper

  afterEach(() => {
    wrapper.destroy()
  })

  it('Header Slot', () => {
    const content = '<p>Header Content</p>'
    wrapper = mount(MainPanel, {
      slots: {
        header: content
      }
    })

    expect(wrapper.html()).toContain(content)
  })

  it('Main Slot', () => {
    const content = '<p>Main Content</p>'
    wrapper = mount(MainPanel, {
      slots: {
        content: content
      }
    })

    expect(wrapper.html()).toContain(content)
  })

  it('Main and Content Slots', () => {
    const mainContent = '<p>Main Content</p>'
    const headerContent = '<p>Header Content</p>'
    wrapper = mount(MainPanel, {
      slots: {
        content: mainContent,
        header: headerContent
      }
    })

    expect(wrapper.html()).toContain(mainContent)
    expect(wrapper.html()).toContain(headerContent)
  })

  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
