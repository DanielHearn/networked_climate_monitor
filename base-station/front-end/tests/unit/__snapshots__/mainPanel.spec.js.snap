// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPanel.vue has the expected html structure 1`] = `
<div
  class="main-panel"
>
  <div
    class="main-panel-header"
  >
    <p>
      Header Content
    </p>
  </div>
   
  <div
    class="main-panel-content"
  >
    <p>
      Main Content
    </p>
  </div>
</div>
`;
